server/protocols/jmap/src/test/java/org/apache/james/jmap/model/message/MimePartTest.java

```diff
@@ -0,0 +1,214 @@
+ * distributed with this work for additional information        *
+ * regarding copyright ownership.  The ASF licenses this file   *
+ * to you under the Apache License, Version 2.0 (the            *
+ * "License"); you may not use this file except in compliance   *
+ * with the License.  You may obtain a copy of the License at   *
+ *                                                              *
+ *   http://www.apache.org/licenses/LICENSE-2.0                 *
+ *                                                              *
+ * Unless required by applicable law or agreed to in writing,   *
+ * software distributed under the License is distributed on an  *
+ * "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY       *
+ * KIND, either express or implied.  See the License for the    *
+ * specific language governing permissions and limitations      *
+ * under the License.                                           *
+ ****************************************************************/
+
+package org.apache.james.jmap.model.message;
+
+import static org.assertj.core.api.Assertions.assertThat;
+
+import org.apache.commons.io.IOUtils;
+import org.junit.Test;
+
+public class MimePartTest {
+
+    @Test
+    public void isHTMLShouldReturnTrueWhenHTMLSubType() {
+        MimePart mimePart = MimePart.builder()
+            .addSubType("html")
+            .build();
+        assertThat(mimePart.isHTML()).isTrue();
+    }
+
+    @Test
+    public void isHTMLShouldReturnFalseWhenOtherSubType() {
+        MimePart mimePart = MimePart.builder()
+            .addSubType("other")
+            .build();
+        assertThat(mimePart.isHTML()).isFalse();
+    }
+
+    @Test
+    public void isPlainShouldReturnTrueWhenPlainSubType() {
+        MimePart mimePart = MimePart.builder()
+            .addSubType("plain")
+            .build();
+        assertThat(mimePart.isPlain()).isTrue();
+    }
+
+    @Test
+    public void isPlainShouldReturnFalseWhenOtherSubType() {
+        MimePart mimePart = MimePart.builder()
+            .addSubType("other")
+            .build();
+        assertThat(mimePart.isPlain()).isFalse();
+    }
+
+    @Test
+    public void retrieveTextHtmlBodyShouldReturnEmptyWhenOtherSubType() {
+        MimePart mimePart = MimePart.builder()
+            .addSubType("other")
+            .build();
+        assertThat(mimePart.retrieveTextHtmlBody()).isEmpty();
+    }
+
+    @Test
+    public void retrieveTextHtmlBodyShouldReturnThisWhenHtmlSubType() {
+        String expectedContent = "<b>content</b>";
+        MimePart htmlMimePart = MimePart.builder()
+                .addMediaType("text")
+                .addSubType("html")
+                .addBodyContent(IOUtils.toInputStream(expectedContent))
+                .build();
+
+        assertThat(htmlMimePart.retrieveTextHtmlBody()).isPresent();
+        assertThat(htmlMimePart.retrieveTextHtmlBody().get()).isEqualTo(expectedContent);
```

Comment:
Every where here I guess you can use asserts on optionnals : 

```
assertThat(htmlMimePart.retrieveTextHtmlBody()).contains(expectedContent);
```

It's more concise and easier to read ;-)